backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/webaudio/WebAudioAPIManager.java

```diff
@@ -0,0 +1,230 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.backends.gwt.webaudio;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.LifecycleListener;
+import com.badlogic.gdx.audio.Music;
+import com.badlogic.gdx.audio.Sound;
+import com.badlogic.gdx.backends.gwt.GwtApplicationConfiguration;
+import com.badlogic.gdx.backends.gwt.preloader.AssetDownloader;
+import com.badlogic.gdx.backends.gwt.GwtApplication;
+import com.badlogic.gdx.files.FileHandle;
+import com.google.gwt.core.client.JavaScriptObject;
+import com.google.gwt.dom.client.CanvasElement;
+import com.google.gwt.media.client.Audio;
+import com.google.gwt.typedarrays.shared.ArrayBuffer;
+import com.google.gwt.typedarrays.shared.Int8Array;
+import com.google.gwt.typedarrays.shared.TypedArrays;
+import com.google.gwt.xhr.client.ReadyStateChangeHandler;
+import com.google.gwt.xhr.client.XMLHttpRequest;
+import com.google.gwt.xhr.client.XMLHttpRequest.ResponseType;
+
+public class WebAudioAPIManager implements LifecycleListener {
+	private final JavaScriptObject audioContext;
+	private final JavaScriptObject globalVolumeNode;
+	private final AssetDownloader assetDownloader;
+	private final AudioControlGraphPool audioControlGraphPool;
+	private static boolean soundUnlocked;
+
+	public WebAudioAPIManager () {
+		this.assetDownloader = new AssetDownloader();
+		this.audioContext = createAudioContextJSNI();
```

Comment:
audioContext creation at this point (without user interaction) may print a warning (see my comment in PR discussion).